jazz_metrics/index.js

```diff
@@ -0,0 +1,319 @@
+  }
+
+  try {
+    /*
+     * event input format :
+     *   {
+     *        "domain": "jazztest",
+     *        "service": "get-monitoring-data",
+     *        "environment": "prod",
+     *        "end_time": "2017-06-27T06:56:00.000Z",
+     *        "start_time": "2017-06-27T05:55:00.000Z",
+     *        "interval":"300",
+     *        "statistics":"average"
+     *    }
+     */
+    var eventBody = event.body;
+    genericValidation(event)
+      .then(() => validateUtils.validateGeneralFields(eventBody))
+      .then(() => getAssetsDetails(config, eventBody))
+      .then(res => validateAssets(res, eventBody))
+      .then(res => getMetricsDetails(res, cloudwatch, config))
+      .then(res => {
+        var finalObj = utils.massageData(res, eventBody);
+        return cb(null, responseObj(finalObj, eventBody));
+      })
+      .catch(error => {
+        if (error.result === "inputError") {
+          return cb(JSON.stringify(errorHandler.throwInputValidationError(error.message)));
+        } else if (error.result === "notFoundError") {
+          return cb(JSON.stringify(errorHandler.throwNotFoundError(error.message)));
+        } else if (error.result === "unauthorized") {
+          return cb(JSON.stringify(errorHandler.throwUnauthorizedError(error.message)));
+        } else {
+          return cb(JSON.stringify(errorHandler.throwInternalServerError("Error in fetching cloudwatch metrics")));
+        }
+      });
+  } catch (e) {
+    return cb(JSON.stringify(errorHandler.throwInternalServerError("Error in fetching cloudwatch metrics")));
+  }
+
+};
+
+function genericValidation(event) {
+  return new Promise((resolve, reject) => {
+    if (!event && !event.body) {
+      reject({
+        result: "inputError",
+        message: "Invalid Input Error"
+      });
+    }
+    if (!event.principalId) {
+      reject({
+        result: "unauthorized",
+        message: "Unauthorized"
+      });
+    }
+
+    resolve();
+  });
+}
+
+function getAssetsDetails(config, eventBody) {
+  return new Promise((resolve, reject) => {
+    var asset_api_payload = {
+      "service": eventBody.service,
+      "domain": eventBody.domain,
+      "environment": eventBody.environment
+    };
+    var asset_api_options = {
+      url: config.SERVICE_API_URL + config.ASSETS_URL,
+      headers: {
+        "Content-Type": "application/json",
+      },
+      method: "POST",
+      rejectUnauthorized: false,
+      requestCert: true,
+      async: true,
+      json: true,
+      body: asset_api_payload,
+    };
+
+    logger.info("asset_api_options :- " + JSON.stringify(asset_api_options));
+
+    request(asset_api_options, (error, response, body) => {
+      if (error) {
+        reject(error);
+      } else {
+        var apiAssetsArray = body.data;
+        if (apiAssetsArray && apiAssetsArray.length === 0) {
+          logger.error("Assets not found for this service, domain, environment. ", JSON.stringify(asset_api_options));
+          resolve([]);
```

Comment:
Why is this resolve and not reject?

Reply:
If no assets are available, then assets params is assigned as empty array in response.